libtiledbsoma/src/soma/soma_array.cc

```diff
@@ -432,6 +432,15 @@ std::vector<int64_t> SOMAArray::shape() {
     return result;
 }
 
+Enumeration SOMAArray::get_enum(std::string name) {
+    return ArrayExperimental::get_enumeration(*ctx_, *arr_, name);
+}
+
+std::optional<std::string> SOMAArray::get_enum_label_on_attr(std::string name) {
+    auto attr = arr_->schema().attribute(name);
+    return AttributeExperimental::get_enumeration_name(*ctx_, attr);
+}
+
```

Comment:
These two are good.  

I keep wondering if we also need
- a 'get me enumerations' given the array, ie something that loops over the attributes and returns them all
- the name mapping is still confusing because attribute name is not necessarily the same as enumeration, but if they differ how do we discover the mapping?
- the second can also we wrapped as a simple `bool` predicate checking whether given a name the return is null or not.

Reply:
I think a function that returns the mapping of attributes with its associated enumeration would be great to add. I'll work on that now.